server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader10to11.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.InstanceId;
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.fate.zookeeper.ZooReaderWriter;
+import org.apache.accumulo.core.fate.zookeeper.ZooUtil;
+import org.apache.accumulo.core.manager.state.tables.TableState;
+import org.apache.accumulo.core.metadata.MetadataTable;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.conf.store.NamespacePropKey;
+import org.apache.accumulo.server.conf.store.PropStore;
+import org.apache.accumulo.server.conf.store.PropStoreKey;
+import org.apache.accumulo.server.conf.store.SystemPropKey;
+import org.apache.accumulo.server.conf.store.TablePropKey;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Upgrader10to11 implements Upgrader {
+
+  private static final Logger log = LoggerFactory.getLogger(Upgrader10to11.class);
+
+  // Included for upgrade code usage any other usage post 3.0 should not be used.
+  private static final TableId REPLICATION_ID = TableId.of("+rep");
+
+  public Upgrader10to11() {
+    super();
+  }
+
+  @Override
+  public void upgradeZookeeper(final ServerContext context) {
+    log.info("upgrade of ZooKeeper entries");
+
+    var zrw = context.getZooReaderWriter();
+    var iid = context.getInstanceID();
+
+    // if the replication base path (../tables/+rep) assume removed or never existed.
+    if (!checkReplicationTableInZk(iid, zrw)) {
+      log.debug("replication table root node does not exist in ZooKeeper - nothing to do");
+      return;
+    }
+
+    // if the replication table is online - stop. There could be data in transit.
+    if (!checkReplicationOffline(iid, zrw)) {
+      throw new IllegalStateException(
+          "Replication table is not offline. Cannot continue with upgrade that will remove replication with replication active");
+    }
+
+    deleteReplicationConfigs(zrw, iid, context.getPropStore());
+
+    deleteReplicationTableZkEntries(zrw, iid);
+
+  }
+
+  @Override
+  public void upgradeRoot(final ServerContext context) {
+    log.info("upgrade root - skipping, nothing to do");
+  }
+
+  @Override
+  public void upgradeMetadata(final ServerContext context) {
+    log.info("upgrade metadata entries");
+    deleteReplMetadataEntries(context);
+  }
+
+  /**
+   * remove +rep entries from metadata.
+   */
+  private void deleteReplMetadataEntries(final ServerContext context) {
+    try (BatchDeleter deleter =
+        context.createBatchDeleter(MetadataTable.NAME, Authorizations.EMPTY, 10)) {
+      deleter.setRanges(Collections.singletonList(
+          new Range(REPLICATION_ID.canonical() + ";", REPLICATION_ID.canonical() + "<")));
+      deleter.delete();
```

Comment:
This deletes the `+rep` table entries, right? Do we also need to delete any replication entries within other tables? Reference: https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java#L404

Reply:
Maybe?  Although I configured replication I did not actually set-up and run two full instances (yet) - so I have not seen those entries. I think removing the replication iterator from the configs is the minimum required to get an upgraded instance to run - there maybe additional artifacts, but as far as I know, they do not impact functionality.

As other entries are identified, they can be added to this PR is done as a follow-on.  As-is this is sufficient to upgrade and run 3.0 code as it stands now (well, that's the goal)

Reply:
I don't know what other tables would have replication entries. There's just the replication table (in its entirety) and these entries in the metadata table, as far as I can tell.

Reply:
The repTableRange would be deleting the metadata entries that define the replication table. The repWalRange would be deleting the additionally tracked WALs that are preserved because they are in use, and intended to prevent the accumulo-gc from deleting the still-referenced WALs. I think that's all we're expecting.